### PR TITLE
Added exercise 5 firmware update implementation

### PR DESCRIPTION
**Pull request by malw4**

### CODE REVIEW

File: docs/academy/exercise5.md
Comment (JZimnol) [inline docs/academy/exercise5.md:446]:
```suggestion
    Does the **State** `/5/*/3` report `0` and the **Update Result** `/5/*/5` report `1`? Congratulations! You've successfully updated the firmware of your device 🎉
```

File: docs/academy/gradproject.md
Comment (JZimnol) [inline docs/academy/gradproject.md:20]: The feedback form does not include the Module 5 Exercise

**Review by JZimnol — COMMENTED** [review]
@malw4 reviewed. Remember to NOT merge this before the newest `pico-client` is released.

File: docs/academy/exercise5.md
Comment (malw4) [reply docs/academy/exercise5.md:7]: sorry, copy and paste gone wrong

File: docs/academy/exercise5.md
Comment (malw4) [reply docs/academy/exercise5.md:238]: then please update the github version

File: docs/academy/exercise5.md
Comment (malw4) [reply docs/academy/exercise5.md:426]: look at this https://iotdevzone.avsystem.com/docs/Coiote_IoT_DM/Single_update/#download-upgrade-process. There is also this note

File: docs/academy/exercise5.md
Comment (JZimnol) [inline docs/academy/exercise5.md:17]:
```suggestion
    This part only describes functions that are in the code in **Anjay-pico-client/firmware_update** directory. The user doesn’t have to modify the code.
```

File: docs/academy/exercise5.md
Comment (JZimnol) [inline docs/academy/exercise5.md:208]: You've indented the code, but not the whole section. It renders badly. Applies for other code snippets.

File: docs/academy/module5.md
Comment (JZimnol) [inline docs/academy/module5.md:77]: No no no, this place is `z dupy` for that in my opinion. THis hsould be in Exercise 5, before `Implement Firmware Update` for example.

File: docs/academy/exercise5.md
Comment (JZimnol) [inline docs/academy/exercise5.md:398]: It is single instance, so no `*`.

**Review by JZimnol — COMMENTED** [review]
@malw4 re-reviewed.

File: docs/academy/exercise5.md
Comment (JZimnol) [inline docs/academy/exercise5.md:212]: The whole code is still indented not the way it should be. The "code section" is OK itself, the code inside is indented unnecessarily.

File: docs/academy/exercise5.md
Comment (JZimnol) [inline docs/academy/exercise5.md:222]:
```suggestion
        void flash_aligned_writer_new(uint8_t *batch_buf,
                                      size_t batch_buf_max_len_bytes,
```

And everywhere else, the code is not aligned after copy-paste.

**Review by JZimnol — COMMENTED** [review]
The code snippets in `firmware_update.c` are still not up to date or are incorrect.

**Review by JZimnol — APPROVED** [review]
@malw4  approving in advance, but please make any necessary changes discussed privately

**Pull request merged**
